src/flamegraph/mod.rs

```diff
@@ -121,7 +123,10 @@ where
         };
 
         let samples = frame.end_time - frame.start_time;
-        let samples_txt = samples.thousands_sep();
+
+        // add thousands separators to `samples`
+        let _ = samples_txt_buffer.write_formatted(&samples, &Locale::en);
```

Comment:
Doesn't this buffer have to be cleared?

Reply:
No. Every time you call `write_formattted` on a `num_format::Buffer` it essentially takes care of clearing.

Reply:
Huh, I wouldn't have guessed from the name, but sounds good!

Reply:
Yah, I should probably add a note in the docs that that's the behavior. I said "essentially" takes clear of clearing, by the way, because it doesn't do the naive thing of clearing bytes to zero or anything like that. Underneath, `Buffer` contains an array and two indices, one indicating the beginning of valid data in the array and the second indicating the end of valid data. "Clearing" just means changing those indices to both point to the end of the array (or both to the beginning - it's the same). Anyway, `write_formatted` takes care to always reset those indices before writing bytes; so this code does the thing you want. Just figured you might appreciate a bit more commentary.

Reply:
As long as it re-uses memory but not content, I'm happy :p